RFC/src/RFC-0132_Merge_Mining_Monero.md

```diff
@@ -0,0 +1,163 @@
+`kernel_mmr_size`,`total_kernel_offset`,`total_script_offset`. Note, this hash does not include the `pow` and `nonce` fields, as these fields are set as part of mining.
+
+To have the chance of mining a Monero block as well as a Tari block, we must obtain a new valid monero block template, by calling [get_block_template].
+This returns a `blocktemplate_blob`, that is, a serialized Monero block containing the Monero block header, coinbase and a list of hashes referencing the 
+transactions included in the block. Additionally, a `blockhashing_blob` is a fixed size blob containing `serialized_monero_header`, `merkle_tree_root` and 
+`txn_count` concatenated together. The `merkle_tree_root` is a _merkle root_ of the coinbase + the transaction hashes contained in the block.
+
+```rust,ignore
+pub struct Block {
+    /// The block header
+    pub header: BlockHeader,
+    /// Coinbase transaction a.k.a miner transaction
+    pub miner_tx: Transaction,
+    /// References to the transactions included in this block
+    pub tx_hashes: Vec<hash::Hash>,
+}
+```
+_fig 1. The Monero block struct_
+
+Next, modify the Monero block template by including the merge mining hash \\( h_m \\) in the extra fields of the coinbase transaction. Monero has a [merge mining subfield] 
+to accommodate this data. Importantly, the extra field data part of the coinbase transaction hash and therefore the `merkle_tree_root`, the `blockhashing_blob` must be 
+reconstructed. A rust port of [Monero's tree hash] algorithm is needed to achieve this. The coinbase hash MUST be the first element to be hashed when constructing the `merkle_tree_root`.
+This satisfies **REQ 2**, proving that the Tari block was constructed before the parent chain's block. 
+
+The block may now be mined. Once a solution is found that satisfies the Tari difficulty, the miner must include enough data to allow the Tari blockchain to assert **REQ 1** and **REQ 2**.
+Concretely, A miner must include the following struct serialized using [bincode] in the `pow` field
+
+```rust,ignore
+pub struct MoneroPowData {
+    /// Monero header fields
+    pub header: MoneroBlockHeader,
+    /// randomX vm key
+    pub key: Vec<u8>,
+    /// transaction count
+    pub transaction_count: u16,
+    /// transaction root
+    pub transaction_root: [u8; 32],
+    /// Coinbase merkle proof hashes
+    pub coinbase_merkle_proof: Vec<[u8; 32]>,
+    /// Coinbase tx from Monero
+    pub coinbase_tx: MoneroTransaction,
+}
+```
+_fig 2. Monero PoW data serialized in Tari blocks
+
+A verifier may now check that the `coinbase_tx` contains the merge mining hash \\( h_m \\), and validate the `coinbase_merkle_proof` against the `transaction_root`.
+The `coinbase_merkle_proof` contains the minimal proof required to construct the `transaction_root`.
+
+For example, a tree with a depth of 2 will have a proof size of 3 * 32 bytes.
```

Comment:
Need to update this to reflect the changes made in upcoming PR - 3 * 32 bytes + 2 u32s used for the depth and path_bitmap